src/main/resources/META-INF/jqassistant-rules/cdi/jakarta-cdi.xml

```diff
@@ -0,0 +1,420 @@
+        <providesConcept refId="cdi:EventProducer"/>
+        <description>Labels all beans declaring "InjectionPoint" of type "jakarta.enterprise.event.Event" with "CDI" and "EventProducer".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (a:Type)-[:DECLARES]->(member:Field:CDI:InjectionPoint),
+              (member)-[:OF_TYPE]->(injectType:Type)
+            WHERE
+              injectType.fqn = "jakarta.enterprise.event.Event"
+            SET
+              a:CDI:EventProducer
+            RETURN
+              DISTINCT a.fqn AS cdiEventProducers
+        ]]></cypher>
+    </concept>
+
+    <concept id="jakarta-cdi:EventConsumer">
+        <providesConcept refId="cdi:EventConsumer"/>
+        <description>Labels all beans declaring method that has parameter of type "jakarta.enterprise.event.Observes" with "CDI" and "EventConsumer".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (a:Type)-[:DECLARES]->(member:Method)-[:HAS]->(param:Parameter),
+              (param)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(injectType:Type)
+            WHERE
+              injectType.fqn = "jakarta.enterprise.event.Observes"
+            SET
+              a:CDI:EventConsumer
+            RETURN
+              DISTINCT a.fqn AS cdiEventConsumer
+        ]]></cypher>
+    </concept>
+
+    <concept id="jakarta-cdi:Named">
+        <providesConcept refId="cdi:Named"/>
+        <description>Labels all types or methods annotated by "jakarta.inject.Named" with "CDI" and "Named".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (e)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(namedType:Type)
+            WHERE
+              namedType.fqn = "jakarta.inject.Named"
+            SET
+              e:CDI:Named
+            RETURN
+              e AS Named
+        ]]></cypher>
+    </concept>
+
+    <concept id="jakarta-cdi:Default">
+        <providesConcept refId="cdi:Default"/>
+        <description>Labels all elements annotated by "jakarta.enterprise.inject.Default" with "CDI" and "Default".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (e)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(defaultType:Type)
+            WHERE
+              defaultType.fqn = "jakarta.enterprise.inject.Default"
+            SET
+              e:CDI:Default
+            RETURN
+              e AS Default
+        ]]></cypher>
+    </concept>
+
+    <concept id="jakarta-cdi:Any">
+        <providesConcept refId="cdi:Any"/>
+        <description>Labels all elements annotated by "jakarta.enterprise.inject.Any" with "CDI" and "Any".
+        </description>
+        <cypher><![CDATA[
+            MATCH
+              (e)-[:ANNOTATED_BY]->()-[:OF_TYPE]->(anyType:Type)
+            WHERE
+              anyType.fqn = "jakarta.enterprise.inject.Any"
+            SET
+              e:CDI:Any
+            RETURN
+              e AS Any
+        ]]></cypher>
+    </concept>
+
```

Comment:
New fehlt (auch, wenn es deprecated ist).

Reply:
Done. Ggf. noch Tests vervollständigen.

Reply:
In Version 3 noch vorhanden, in Version 4 anscheinend nicht mehr. Wollen wir die Regel trotzdem drinlassen? Dann können wir sie wahrscheinlich nicht testen.
https://jakarta.ee/specifications/cdi/3.0/apidocs/jakarta/enterprise/inject/new

Reply:
Würde ich drin lassen, falls das mal in einem Projekt auftaucht.